targets/TARGET_STM/TARGET_STM32H7/i2c_device.c

```diff
@@ -0,0 +1,433 @@
+  uint32_t sclh;       /* SCL high period */
+  uint32_t scll;       /* SCL low period */
+} I2C_Timings_t;
+/**
+  * @}
+  */
+
+/** @defgroup I2C_DEVICE_Private_Constants I2C_DEVICE Private Constants
+  * @{
+  */
+static const I2C_Charac_t I2C_Charac[] =
+{
+  [I2C_SPEED_FREQ_STANDARD] =
+  {
+    .freq = 100000,
+    .freq_min = 80000,
+    .freq_max = 120000,
+    .hddat_min = 0,
+    .vddat_max = 3450,
+    .sudat_min = 250,
+    .lscl_min = 4700,
+    .hscl_min = 4000,
+    .trise = 640,
+    .tfall = 20,
+    .dnf = I2C_DIGITAL_FILTER_COEF,
+  },
+  [I2C_SPEED_FREQ_FAST] =
+  {
+    .freq = 400000,
+    .freq_min = 320000,
+    .freq_max = 480000,
+    .hddat_min = 0,
+    .vddat_max = 900,
+    .sudat_min = 100,
+    .lscl_min = 1300,
+    .hscl_min = 600,
+    .trise = 250,
+    .tfall = 100,
+    .dnf = I2C_DIGITAL_FILTER_COEF,
+  },
+  [I2C_SPEED_FREQ_FAST_PLUS] =
+  {
+    .freq = 1000000,
+    .freq_min = 800000,
+    .freq_max = 1200000,
+    .hddat_min = 0,
+    .vddat_max = 450,
+    .sudat_min = 50,
+    .lscl_min = 500,
+    .hscl_min = 260,
+    .trise = 60,
+    .tfall = 100,
+    .dnf = I2C_DIGITAL_FILTER_COEF,
+  },
+};
+/**
+  * @}
+  */
+
+/** @defgroup I2C_DEVICE_Private_Variables I2C_DEVICE Private Variables
+* @{
+*/
+static I2C_Timings_t I2c_valid_timing[I2C_VALID_TIMING_NBR];
+static uint32_t      I2c_valid_timing_nbr = 0;
+/**
+  * @}
+  */
+
+/** @defgroup I2C_DEVICE_Private_Functions I2C_DEVICE Private Functions
+  * @{
+  */
+/**
+  * @brief  Compute PRESC, SCLDEL and SDADEL.
+  * @param  clock_src_freq I2C source clock in HZ.
+  * @param  I2C_speed I2C frequency (index).
+  * @retval None.
+  */
+static void I2C_Compute_PRESC_SCLDEL_SDADEL(uint32_t clock_src_freq, uint32_t I2C_speed)
+{
+  uint32_t prev_presc = I2C_PRESC_MAX;
+  uint32_t ti2cclk;
+  int32_t  tsdadel_min, tsdadel_max;
+  int32_t  tscldel_min;
+  uint32_t presc, scldel, sdadel;
+  uint32_t tafdel_min, tafdel_max;
+
+  ti2cclk   = (SEC2NSEC + (clock_src_freq / 2U))/ clock_src_freq;
+
+  tafdel_min = (I2C_USE_ANALOG_FILTER == 1U) ? I2C_ANALOG_FILTER_DELAY_MIN : 0U;
+  tafdel_max = (I2C_USE_ANALOG_FILTER == 1U) ? I2C_ANALOG_FILTER_DELAY_MAX : 0U;
+
+  /* tDNF = DNF x tI2CCLK
+     tPRESC = (PRESC+1) x tI2CCLK
+     SDADEL >= {tf +tHD;DAT(min) - tAF(min) - tDNF - [3 x tI2CCLK]} / {tPRESC}
+     SDADEL <= {tVD;DAT(max) - tr - tAF(max) - tDNF- [4 x tI2CCLK]} / {tPRESC} */
+
+  tsdadel_min = (int32_t)I2C_Charac[I2C_speed].tfall + (int32_t)I2C_Charac[I2C_speed].hddat_min -
+    (int32_t)tafdel_min - (int32_t)(((int32_t)I2C_Charac[I2C_speed].dnf + 3) * (int32_t)ti2cclk);
+
+  tsdadel_max = (int32_t)I2C_Charac[I2C_speed].vddat_max - (int32_t)I2C_Charac[I2C_speed].trise -
+    (int32_t)tafdel_max - (int32_t)(((int32_t)I2C_Charac[I2C_speed].dnf + 4) * (int32_t)ti2cclk);
+
+
+  /* {[tr+ tSU;DAT(min)] / [tPRESC]} - 1 <= SCLDEL */
+  tscldel_min = (int32_t)I2C_Charac[I2C_speed].trise + (int32_t)I2C_Charac[I2C_speed].sudat_min;
+
+  if (tsdadel_min <= 0)
+  {
```

Comment:
Can you run this file through astyle please ? the style is different than the rest of HAL implementation here.

Reply:
Should be ok now. Thanks.